pkg/action/upgrade.go

```diff
@@ -57,3 +57,19 @@ func (i *Upgrade) Name(chart *chart.Chart, args []string) (string, error) {
 	// If we dont have our annotations then return the base name
 	return chart.Metadata.Name, nil
 }
+
+// SetNamespace sets the Namespace that should be used in action.Upgrade
+//
+// This will read the chart annotations. If no annotations, it leave the existing ns in the action.
+func (i *Upgrade) SetNamespace(chart *chart.Chart, defaultns string) {
```

Comment:
We should add unit tests for this new function. E.g: https://github.com/rancher-sandbox/hypper/blob/viccuad%2Fadd-shared-deps/pkg/action/dependency_shared_test.go#L81-L101. The different TestSetNamespace() functions will clash, so either just rename the TestSetNamespace() to TestUpgradeSetNamespace(), or rename the others (as done here, but then I may need to rebase https://github.com/rancher-sandbox/hypper/pull/80/commits/a0acead55d686354d1a38ce5d9eb7128b8bbc46a#diff-b97a6f3723149f75e7945127b215944ea6a752bcf41a2e1c00b65ae0a409bc7cR34-R38).

Reply:
good point, didnt even think about the name clash :o

Reply:
unfortunately as I want to reuse some of your added functions I will have to wait for your Pr to merge :D

Reply:
done!